lib/saml/settings_service.rb

```diff
@@ -3,14 +3,22 @@ module SAML
   # This class is responsible for putting together a complete ruby-saml
   # SETTINGS object, meaning, our static SP settings + the IDP settings
   # which must be fetched once and only once via IDP metadata.
-  require 'singleton'
   class SettingsService
-    include Singleton
-
     attr_reader :saml_settings
 
+    METADATA_URI = URI(SAML_CONFIG['metadata_url'])
+
     def initialize
-      @saml_settings ||= fetch_idp_metadata
+      @saml_settings = create_settings_from_metadata
+    end
+
+    def self.metadata
+      return @metadata if defined?(@metadata)
+      http = Net::HTTP.new(METADATA_URI.host, METADATA_URI.port)
```

Comment:
Wont `METADATA_URI.host` & `METADATA_URI.port` spit a `NoMethodError`?  `METADATA_URI` is just a string